lib/types.nix

```
@@ -373,6 +373,14 @@ rec {
       merge = mergeEqualOption;
     };
 
+    hex = mkOptionType {
+      name = "hex";
+      description = "hexadecimal number";
```

Is it really a number if it's in a string?
This might feel like splitting hairs, but it will save some trial and error when the `description` is all available documentation.

```suggestion
      description = "hexadecimal digits string";
```

This also implicitly excludes the prefixes that indicate the start of a hexadecimal number, such as `0x` or `#`.

---

And maybe it should be called `types.hexStr`